go/common/node/node.go

```diff
@@ -71,8 +72,9 @@ type Node struct { // nolint: maligned
 	// Beacon contains information for this node's participation
 	// in the random beacon protocol.
 	//
-	// NOTE: This is reserved for future use.
-	Beacon cbor.RawMessage `json:"beacon,omitempty"`
+	// TODO: This is optional for now, make mandatory once enough
+	// nodes provide this field.
+	Beacon *BeaconInfo `json:"beacon,omitempty"`
```

Comment:
This probably needs to stay unchanged as well.

Reply:
The more I think about this, the more I'm vehemently against adding something that sticks on-chain data that can't be validated.

Reply:
> The more I think about this, the more I'm vehemently against adding something that sticks on-chain data that can't be validated.

But that's already there. If nodes have invalid data in it at the time of the upgrade, they will be removed from genesis.